src/tls_openssl.c

```diff
@@ -153,6 +153,11 @@ tls_t *tls_new(xmpp_conn_t *conn)
         if (tls->ssl == NULL)
             goto err_free_ctx;
 
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
```

Comment:
According to https://man.openbsd.org/SSL_CTX_set_tlsext_servername_callback.3 the function appeared in OpenSSL 0.9.8f. I guess `0x10000000L` should be different to cover 0.9.x. Plus as far as I remember, openssl changed version convention at some point -- need to check this too.

Reply:
How about this?

`#if OPENSSL_VERSION_NUMBER >= 0x0908060L && !defined(OPENSSL_NO_TLSEXT)`

Reply:
Looks good to me. Please, update the pull request.

Reply:
Done!